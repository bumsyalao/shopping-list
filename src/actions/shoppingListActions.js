

 export const setBudget = (budget) => dispatch => {
   dispatch({
     type: 'SET_BUDGET',
     payload: budget
   })
 };

export const addItem = (Items) => dispatch => {
  dispatch({
    type: 'ADD_ITEM',
    newItem: Items,
  })
};

export const getItems = () => dispatch => {
  dispatch({
    type: 'GET_ITEMS'
  })
};

