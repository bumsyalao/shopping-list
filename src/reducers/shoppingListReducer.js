let initialState = {
  budget: null,
  items: [],
}
export default (state = initialState, action) => {
  switch (action.type) {
   case 'SET_BUDGET':
    return {
      ...state,
     budget: action.payload
    }
    
   case 'ADD_ITEM':
    let newState = {...state}
     newState.items.push(action.newItem);
     return newState

   case 'GET_ITEMS':
     return state.items

   default:
    return state
  }
 };
