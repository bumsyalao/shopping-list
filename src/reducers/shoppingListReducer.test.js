import shoppingListReducer, { initialState } from './shoppingListReducer';
import { setBudget, addItem, getItems } from '../actions/shoppingListActions';

const mybudget = 5000;

describe('shoppingList reducer', () => {
  it('return default state', () => {
    expect(shoppingListReducer(undefined, {})).toMatchSnapshot();
  });

  it('should set budget when set budget is called', () => {
    const newState = shoppingListReducer(initialState, {
      type: 'SET_BUDGET',
      payload: mybudget,
    });
    expect(newState.budget).toMatchSnapshot();
    expect(newState.budget).toEqual(5000);

  });

  it('should add item when addItem is called', () => {
    const newState = shoppingListReducer(initialState, {
      type: 'ADD_ITEM',
      newItem: {
        product: 'shoes',
        quantity: '3',
        price: '300'
      }
    });

    expect(newState.items.product).toMatchSnapshot();
    expect(newState.items[0].product).toEqual('shoes');
  });

  it('should get items', () => {
    const newState = shoppingListReducer(initialState, {
      type: 'GET_ITEMS'
    });

    expect(newState.items).toMatchSnapshot();
  })

});