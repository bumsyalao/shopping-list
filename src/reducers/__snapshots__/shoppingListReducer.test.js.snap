// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoppingList reducer return default state 1`] = `
Object {
  "budget": null,
  "items": Array [],
}
`;

exports[`shoppingList reducer should add item when addItem is called 1`] = `undefined`;

exports[`shoppingList reducer should get items 1`] = `undefined`;

exports[`shoppingList reducer should set budget when set budget is called 1`] = `5000`;
